aws/resource_aws_transfer_server.go

```diff
@@ -0,0 +1,235 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/transfer"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsTransferServer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsTransferServerCreate,
+		Read:   resourceAwsTransferServerRead,
+		Update: resourceAwsTransferServerUpdate,
+		Delete: resourceAwsTransferServerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"invocation_role": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateArn,
+			},
+
+			"url": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"identity_provider_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  transfer.IdentityProviderTypeServiceManaged,
+				ValidateFunc: validation.StringInSlice([]string{
+					transfer.IdentityProviderTypeServiceManaged,
+					transfer.IdentityProviderTypeApiGateway,
+				}, false),
+			},
+
+			"logging_role": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateArn,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsTransferServerCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).transferconn
+	tags := tagsFromMapTransferServer(d.Get("tags").(map[string]interface{}))
+
+	createOpts := &transfer.CreateServerInput{
+		Tags: tags,
```

Comment:
i think that this is the line that's causing the tags to be "required" when they're supposed to be optional. if we provide the `Tags` attribute in the CreateServerInput call, they cannot be empty.